src/parser.rs

```diff
@@ -286,37 +283,30 @@ impl<'a> Parser<'a> {
         Err("parse_string: no closing '\"'")
     }
 
-    pub(crate) fn parse_token(&mut self) -> SFVResult<String> {
+    fn parse_token(&mut self, first_char: u8) -> String {
```

Comment:
I don't think passing the first_char as an argument is really necessary.
It only introduces another chance for extra bugs in the future, such as if `first_char` is not actually `input.next()`